src/components/TransactionHistoryRows/TransactionMethod.tsx

```diff
@@ -116,11 +117,18 @@ export const TransactionMethod = ({
             txMetadataWithAssetId?.assetId &&
             txMetadataWithAssetId?.value && (
               // TODO(gomes): add isTransactionMetadata type guard
-              <ApprovalAmount
-                assetId={txMetadataWithAssetId.assetId}
-                value={txMetadataWithAssetId.value}
-                parser={txMetadataWithAssetId.parser}
-              />
+              <ErrorBoundary
```

Comment:
Was there an error at runtime which `<ErrorBoundary />` caught?

As you've added new checks for `if (!approvedAsset || !approvedAssetMarketData)`, all grounds should now be covered:
- if we have an asset, we have the symbol/precision needed to display the amount
- if we have market data, we are able to get the max/supply to maybe display infinite approvals
- Else if any of these two is missing, we return default translations

I don't think any code path in  `makeAmountOrDefault` would actually throw here 👀

Reply:
> Was there an error at runtime which <ErrorBoundary /> caught?

The crash was caused by a custom error manually thrown if the asset was not found (see the #3774 issue for the error screenshot and code location). So I figured we could use an ErrorBoundary to handle it for that specific component, but @0xdef1cafe said he wasn't a fan. So I've removed the "throw" and dealt with it while attempting to format it. It might be confusing because my first commit was rolledback completely by the second... but the error throwing existed already (and now it will be removed if the PR is merged). As you said, since the case it was intercepting is dealt with later in a condition it shouldn't be an issue though!

Reply:
@firebomb1 if you remove the `<ErrorBoundary />` now that it's no longer needed, i think there's 3 of us that are happy to stamp this with the improved tx parsing UX :)

Reply:
Oh my bad, I thought I had removed it in [e7fae96](https://github.com/shapeshift/web/pull/3778/commits/e7fae968a55eb9aa0fe127f0e00c52b9de97ba8e) (hence my previous comment) but apparently I didn't commit the changes to TransactionMethod, I will fix this now.